spring-cloud-aws-sqs/src/test/java/io/awspring/cloud/sqs/integration/SqsIntegrationTests.java

```diff
@@ -399,6 +415,18 @@ void listen(Message<String> message, MessageHeaders headers, Acknowledgement ack
 		}
 	}
 
+	static class MaxConcurrentMessagesListener {
+
+		@Autowired
+		LatchContainer latchContainer;
+
+		@SqsListener(queueNames = MAX_CONCURRENT_MESSAGES_QUEUE_NAME, maxMessagesPerPoll = "1", maxConcurrentMessages = "10", id = "max-concurrent-messages")
```

Comment:
Can we have this test with maxMessagesPerPoll to 10, maxConcurrentMessages to 20 and 20 messages please?

That would represent what I believe to be a more common scenario.

Reply:
That makes sense. I have fixed it.
It seemed that the test would not pass unless the list of messages was sent twice, so that is what I did.